source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -415,6 +447,60 @@ void ProcessorState::continueIfNecessary() {
   }
 }
 
+bool ProcessorState::handleStreamedBodyResponse(const CommonResponse& common_response) {
+  Buffer::OwnedImpl chunk_data;
+  auto chunk = dequeueStreamingChunk(chunk_data);
+  ENVOY_BUG(chunk != nullptr, "Bad streamed body callback state");
+  if (common_response.has_body_mutation()) {
+    ENVOY_LOG(debug, "Applying body response to chunk of data. Size = {}", chunk->length);
+    MutationUtils::applyBodyMutations(common_response.body_mutation(), chunk_data);
+  }
+  bool should_continue = chunk->end_stream;
+  if (chunk_data.length() > 0) {
+    ENVOY_LOG(trace, "Injecting {} bytes of data to filter stream", chunk_data.length());
+    injectDataToFilterChain(chunk_data, chunk->end_stream);
+  }
+
+  if (queueBelowLowLimit()) {
+    clearWatermark();
+  }
+  if (chunk_queue_.empty()) {
```

Comment:
can you factor lines 467-471 out along with 492-499 to a helper function? They are not identical but easily parameterized. E.g.

```
onFinish(chunk_queue_.empty(), callback_state_);
```
here and
```
onFinish(end_of_stream, CallbackState::StreamedBodyCallback);
```
below, with impl
```
void onFinish(bool end_of_stream, CallbackState state) {
 if (end_of_stream) {
    onFinishProcessorCall(Grpc::Status::Ok);
  } else {
    onFinishProcessorCall(Grpc::Status::Ok, state);
  }
}
```

I'm guessing there are other occurrences if this pattern but I didn't look for them.